lib/bolt/cli.rb

```diff
@@ -456,7 +458,13 @@ def show_plan(plan_name)
 
     def list_plans
       plans = pal.list_plans
-      plans.select! { |plan| plan.first.include?(options[:filter]) } if options[:filter]
+      if options[:filter] || config.project.plans.any?
+        plans.select! do |plan|
+          filter = options[:filter] ? plan.first.include?(options[:filter]) : true
+          in_project = config.project.plans.any? ? config.project.plans.include?(plan.first) : true
+          filter && in_project
+        end
+      end
```

Comment:
Same here as with `list_tasks`

Reply:
Oh my gosh my brain has clearly left for the weekend.